mlbusinesscomponents/src/main/java/com/mercadolibre/android/mlbusinesscomponents/components/touchpoint/view/cover_carousel/CoverCardViewPagerAdapter.java

```diff
@@ -0,0 +1,77 @@
+package com.mercadolibre.android.mlbusinesscomponents.components.touchpoint.view.cover_carousel;
+
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.viewpager.widget.PagerAdapter;
+import com.mercadolibre.android.mlbusinesscomponents.components.touchpoint.callback.OnClickCallback;
+import com.mercadolibre.android.mlbusinesscomponents.components.touchpoint.domain.model.cover_carousel.model.cover_card.CoverCardInterface;
+import com.mercadolibre.android.mlbusinesscomponents.components.touchpoint.view.cover_carousel.cover_card.CoverCardView;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CoverCardViewPagerAdapter extends PagerAdapter {
+
+    private final Context context;
+    @Nullable private OnClickCallback onClickCallback;
+    private final List<CoverCardView> elementsView;
+
+    /* default */ CoverCardViewPagerAdapter(final Context context) {
+        this.context = context;
+        elementsView = new ArrayList<>();
+    }
+
+    /* default */ void setElementsView(final List<CoverCardInterface> itemsView) {
+        final List<CoverCardView> coverCardsViews = new ArrayList<>();
+
+        CoverCardView view;
+        for (final CoverCardInterface itemData : itemsView) {
+            view = new CoverCardView(context);
+            view.setOnClickCallback(onClickCallback);
+            view.bind(itemData);
+            coverCardsViews.add(view);
+        }
+
+        elementsView.clear();
```

Comment:
Y si en vez de hacer un clear de todas las view las reutilizamos? creo que en el swipe to refresh se va ver raro este clear